source/extensions/filters/http/peer_metadata/filter.cc

```diff
@@ -323,13 +333,37 @@ void FilterConfig::setFilterState(StreamInfo::StreamInfo& info, bool downstream,
 
 Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers, bool) {
   config_->discoverDownstream(decoder_callbacks_->streamInfo(), headers);
-  config_->injectUpstream(headers);
+  config_->injectUpstream(decoder_callbacks_->streamInfo(), headers);
   return Http::FilterHeadersStatus::Continue;
 }
 
+bool MXPropagationMethod::skipMXHeaders(const StreamInfo::StreamInfo& info) const {
+  const auto& cluster_info = info.upstreamClusterInfo();
+  if (cluster_info && cluster_info.value()) {
+    const auto& cluster_name = cluster_info.value()->name();
+    if (cluster_name == "PassthroughCluster") {
+      return true;
+    }
+    const auto& filter_metadata = cluster_info.value()->metadata().filter_metadata();
+    const auto& it = filter_metadata.find("istio");
+    if (it != filter_metadata.end()) {
+      const auto& skip_mx = it->second.fields().find("external");
+      if (skip_mx != it->second.fields().end()) {
+        const auto skip_mx_value = skip_mx->second.bool_value();
+        if (skip_mx_value) {
+          return true;
+        }
+      }
+    } else {
+      return true;
```

Comment:
This should be `false`? It doesn't have istio metadata.

Reply:
done